### DISCUSSION TITLE
forwarding_rules not working?

### DISCUSSION THREAD

**Discussion by zfLQ2qx2** [DNSCrypt/dnscrypt-proxy#1922] [Q&A]
I am trying to forward all queries for onion addresses to tor's resolver running on 127.0.0.1:9053, with all other addresses being directed to a DOH resolver. I've uncommented forwarding_rules in dnscrypt-proxy.toml and provided the full path to forwarding_rules.txt, added entries for "onion 127.0.0.1:9053" and "exit 127.0.0.1:9053" to the forwarding_rules.txt, and I've enabled AutomapHostsOnResolve in torrc. There are no complaints from dnscrypt-proxy on restart however all of the onion addresses are still being sent to the DOH servers for resolution.

**Comment by jedisct1** [comment]
Do you ever get an IP address back for `.onion` domains?

Tor doesn't use DNS (well, it can, but that happens on the exit nodes, not on your computer). `.onion` are not regular domains.
